Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 22471)
+++ CMakeLists.txt	(working copy)
@@ -884,7 +884,7 @@
   IF (EXISTS "${HDF5_SOURCE_DIR}/release_docs" AND IS_DIRECTORY "${HDF5_SOURCE_DIR}/release_docs")
     SET (release_files
         ${HDF5_SOURCE_DIR}/release_docs/CMake.txt
-        ${HDF5_SOURCE_DIR}/release_docs/Using_CMake.txt
+        ${HDF5_SOURCE_DIR}/release_docs/USING_CMake.txt
         ${HDF5_SOURCE_DIR}/release_docs/COPYING
         ${HDF5_SOURCE_DIR}/release_docs/HISTORY-1_8.txt
         ${HDF5_SOURCE_DIR}/release_docs/INSTALL
